tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskImpl.java

```diff
@@ -484,6 +493,18 @@ public TezCounters getCounters() {
     }
   }
 
+  private void maybeMergeAllTaskAttemptCounters(TezCounters taskCounters) {
```

Comment:
nit
I think we can ditch maybe from the name & just have ```mergeAllTaskAttemptCounters(```, if stateIntervalMontior is enabled it does merge always